### DISCUSSION TITLE
apolloconfig Product Architecture And RoadMap plan

### DISCUSSION THREAD

**Discussion by Anilople** [apolloconfig/apollo#4875] [Ideas]
Maybe we need a product architecture about apolloconfig introduction?

### Product Architecture

I write some in excalidraw, see https://excalidraw.com/#room=bffb6d339355d1e34d7d,JTHEVvt44OF3ZX85IX_PeQ

![image](https://github.com/apolloconfig/apollo/assets/15523186/cd87efd0-e746-42fa-b3a8-4f94a968a82d)

We can add more detail about it, and the plan of feature?

### RoadMap plan

see https://github.com/apolloconfig/apollo/projects/1 for detail

[Contributing to apollo](https://www.apolloconfig.com/#/zh/contributing)

```mermaid
mindmap
  root((RoadMap))
    (OpenApi)
      write Multiple items
      Full manange function
      Command Line Tool
      notfiy by email
    (Observability)
      integration cat,jmx
      more state
    (UI)
      tree, table
      config value input tool
    (performance)
         config-client read 10000 namespaces
         apollo-portal manage 10000 namespace
    (Operational indicators)
        application
        namespace
        config items
        relealse count
```

**Comment by nobodyiam** [comment]
This is a great idea!
